homeassistant/components/ssdp/__init__.py

```diff
@@ -85,56 +87,60 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class _HaServiceInfoDescription(TypedDict, total=True):
+@dataclass
+class _HaServiceInfoDescription:
     """Keys added by HA."""
 
-    x_homeassistant_matching_domains: set[str]
+    matching_domains: set[str]
 
 
-class _SsdpDescriptionBase(TypedDict, total=True):
-    """Compulsory keys for SSDP info."""
+@dataclass
+class SsdpDescription:
+    """Keys for SSDP info."""
 
     ssdp_usn: str
     ssdp_st: str
+    ssdp_location: str | None = None
```

Comment:
This being optional is good. The location won't always be available in ssdp:byebye messages, which are potentially handled by functions of type `SsdpCallback` registered via `ssdp.async_register_callback`.